src/dns_conf.c

```diff
@@ -573,11 +587,35 @@ static int _conf_domain_rule_ipset(char *domain, const char *ipsetname)
 {
 	struct dns_ipset_rule *ipset_rule = NULL;
 	const char *ipset = NULL;
+	char *copy_name = NULL;
 
 	/* Process domain option */
-	if (strncmp(ipsetname, "-", sizeof("-")) != 0) {
+	if (strncmp(ipsetname, "-", sizeof("-")) == 0) {
+		if (_config_domain_rule_flag_set(domain, DOMAIN_FLAG_IPSET_IGNORE) !=
+			0) {
+			goto errout;
+		}
+		return 0;
+	}
+	copy_name = strdup(ipsetname);
+	for (char *tok = strtok(copy_name, ","); tok != NULL;
+		 tok = strtok(NULL, ",")) {
+		enum domain_rule type = DOMAIN_RULE_IPSET_IPV4;
```

Comment:
May be combine two rules into one is better, process them when adding ipset

Reply:
But `dns_domain_rule` only supports one object. Either use two types, namely `DOMAIN_RULE_IPSET_IPV4` and `DOMAIN_RULE_IPSET_IPV6` nor change `struct dns_ipset_rule` to have two members `ipsetname_v4` and `ipsetname_v6`. Do you prefer the latter one? I prefer the former one because the latter one will consume more memory.